File: ExercismFSharp/ExercismFSharp/Allergies.fs
module Allergies

type Allergen =
    | Cats
    | Chocolate
    | Eggs
    | Peanuts
    | Pollen
    | Shellfish
    | Strawberries
    | Tomatoes
    
let allergicTo allergen score =
    let value =
        match allergen with
        | Allergen.Eggs         -> 1
        | Allergen.Peanuts      -> 2
        | Allergen.Shellfish    -> 4
        | Allergen.Strawberries -> 8
        | Allergen.Tomatoes     -> 16
        | Allergen.Chocolate    -> 32
        | Allergen.Pollen       -> 64
        | Allergen.Cats         -> 128
    score &&& value = value 

let allergies score =
    [
        Allergen.Eggs;
        Allergen.Peanuts;
        Allergen.Shellfish;
        Allergen.Strawberries;
        Allergen.Tomatoes;
        Allergen.Chocolate;
        Allergen.Pollen;
        Allergen.Cats
     ]
     |> List.filter
        (fun a -> allergicTo a score)

File: Exercism/python/allergies/allergies.py
class Allergies(object):

    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        item_score = {
            "eggs": 1,
            "peanuts": 2,
            "shellfish": 4,
            "strawberries": 8,
            "tomatoes": 16,
            "chocolate": 32,
            "pollen": 64,
            "cats": 128
        }

        return self.score & item_score[item] != 0

    @property
    def lst(self):
        allergens = [
            (1, "eggs"),
            (2, "peanuts"),
            (4, "shellfish"),
            (8, "strawberries"),
            (16, "tomatoes"),
            (32, "chocolate"),
            (64, "pollen"),
            (128, "cats")
        ]

        result = []

        for value, allergen in allergens:
            if self.score & value != 0:
                result.append(allergen)

        return result
